src/main/java/org/fogbowcloud/sebal/SEBALHelper.java

```diff
@@ -282,9 +282,29 @@ public static String getStationData(Properties properties, Product product, int
 		double latitude = Double.valueOf(String.format("%.10g%n", geoPos.getLat()));
 		double longitude = Double.valueOf(String.format("%.10g%n", geoPos.getLon()));
 
+		String sceneCenterTime = SEBALHelper.getSceneCenterTime(product);
+
 		WeatherStation station = new WeatherStation(properties);
 		UTC startTime = product.getStartTime();
-		return station.getStationData(latitude, longitude, startTime.getAsDate());
+		return station.getStationData(latitude, longitude, startTime.getAsDate(), sceneCenterTime);
+	}
+
+	public static String getSceneCenterTime(Product product) {
```

Comment:
I'm afraid we cannot trust the input data we use (in particular, data coming from USGS and friends). So, we should check the availability of the info we process. e.g "getElement("L1_METADATA_FILE").getElement("PRODUCT_METADATA") and return proper error signals if not